nifi-nar-bundles/nifi-standard-services/nifi-lookup-services-bundle/nifi-lookup-services/src/main/java/org/apache/nifi/lookup/RestLookupService.java

```diff
@@ -0,0 +1,390 @@
+        if (sslService != null) {
+            builder.sslSocketFactory(sslContext.getSocketFactory());
+        }
+
+        client = builder.build();
+
+        String path = context.getProperty(RECORD_PATH).isSet() ? context.getProperty(RECORD_PATH).getValue() : null;
+        if (!StringUtils.isBlank(path)) {
+            recordPath = RecordPath.compile(path);
+        }
+
+        getHeaders(context);
+    }
+
+    private Map<String, String> headers;
+    private void getHeaders(ConfigurationContext context) {
+        headers = new HashMap<>();
+        for (PropertyDescriptor descriptor : context.getProperties().keySet()) {
+            if (descriptor.getName().startsWith("header.")) {
+                headers.put(
+                    descriptor.getDisplayName(),
+                    context.getProperty(descriptor).evaluateAttributeExpressions().getValue()
+                );
+            }
+        }
+    }
+
+    private void setProxy(OkHttpClient.Builder builder) {
+        ProxyConfiguration config = proxyConfigurationService.getConfiguration();
+        if (!config.getProxyType().equals(Proxy.Type.DIRECT)) {
+            final Proxy proxy = config.createProxy();
+            builder.proxy(proxy);
+
+            if (config.hasCredential()){
+                builder.proxyAuthenticator((route, response) -> {
+                    final String credential= Credentials.basic(config.getProxyUserName(), config.getProxyUserPassword());
+                    return response.request().newBuilder()
+                            .header("Proxy-Authorization", credential)
+                            .build();
+                });
+            }
+        }
+    }
+
+    @Override
+    public Optional<Record> lookup(Map<String, Object> coordinates) throws LookupFailureException {
+        final String endpoint = (String)coordinates.get(ENDPOINT_KEY);
+        final String mimeType = (String)coordinates.get(MIME_TYPE_KEY);
+        final String method   = (String)coordinates.get(METHOD_KEY);
+        final String body     = (String)coordinates.get(BODY_KEY);
+
+        if (StringUtils.isBlank(body) && (method.equals("post") || method.equals("put"))) {
+            throw new LookupFailureException(
+                String.format("Used HTTP verb %s without specifying the %s key to provide a payload.", method, BODY_KEY)
+            );
+        }
+
+        Request request = buildRequest(mimeType, method, body, endpoint);
+        try {
+            Response response = executeRequest(request);
+            InputStream is = response.body().byteStream();
+
+            Record record = handleResponse(is, coordinates);
+
+            return record != null
+                    ? Optional.of(record)
+                    : Optional.empty();
+        } catch (Exception e) {
+            getLogger().error("Could not execute lookup.", e);
```

Comment:
Is generally considered 'bad practice' to log an Exception, then re-throw or wrap-and-throw. Would just throw new LookupFailureException(e)

Reply:
Done.